src/coord/src/coord.rs

```diff
@@ -2503,8 +2506,10 @@ impl Coordinator {
             }
         } else {
             // Use the earliest time that is still valid for all sources.
-            let (index_ids, _indexes_complete) = self.catalog.nearest_indexes(&[source_id]);
-            self.indexes.least_valid_since(index_ids)
+            let (index_ids, source_ids) = self.catalog.nearest_indexes(&[source_id]);
+            let mut since = self.indexes.least_valid_since(index_ids);
+            since.join_assign(&self.sources.least_valid_since(source_ids));
```

Comment:
This is getting (more) complicated. With this change, `self.sources` needs to also have the unmaterialized sources (which you added in `ship_sources` above). My understanding is that those will always have their since stay at 0 right? So could we hard code that here instead?

I also have the question then: I thought sources were by definition unmaterialized, so what exactly is update_upper doing with sources anyway?